clang/lib/Lex/DependencyDirectivesScanner.cpp

```diff
@@ -206,6 +206,24 @@ static void skipOverSpaces(const char *&First, const char *const End) {
     ++First;
 }
 
+// Move back by one character, skipping escaped newlines (backslash + \n)
+static char previousChar(const char *First, const char *&Current) {
+  assert(Current > First);
+  --Current;
+  while (Current > First + 1 && isVerticalWhitespace(*Current)) {
```

Comment:
The Lexer allows trailing horizontal whitespace after the `\` (with a warning).  Maybe we should make `Lexer::getEscapedNewLineSize` public and use that here to keep it in sync?